pkg/agents/team6/behaviours.go

```diff
@@ -0,0 +1,58 @@
+package team6
+
+import (
+	"math"
+	// "math/rand"
+)
+
+func (a *CustomAgent6) updateBehaviour() {
+
+	aConf := a.config
+	behaviourMax, behaviourMin := a.getBehaviourRange()
+
+	hpScore := 1 - float64(a.HP())/100.0 //map (minHP,maxHP) -> (1,0)
+	floor := a.Floor() + 1
+	// floor := rand.Intn(600) //* 600 //use for debug
+
+	if floor > a.maxFloorGuess {
+		a.maxFloorGuess = floor + 1
+	}
+
+	fdNum := float64(floor) / float64(a.maxFloorGuess) * aConf.lambda
+	fdDen := float64(aConf.lambda)
+	floorScore := math.Exp(fdNum) / math.Exp(fdDen)
+
+	behaviourParams := []float64{hpScore, floorScore}
+
+	weights := aConf.paramWeights
+	behaviourPrediction := 0.0
+
+	for i := range weights {
+		behaviourPrediction += behaviourParams[i] * weights[i]
+	}
+
+	updateDir := behaviour(behaviourPrediction)*aConf.maxBehaviourThreshold - a.currBehaviour //find direction we need to move in to reach new behaviour prediction
+	updateMag := updateDir * behaviour(1-aConf.stubbornness)                                  //scale movement by stubbornness (minStubborn, maxStubborn) -> (fullMovement, 0)
+	newBehaviour := a.currBehaviour + updateMag
+
+	if newBehaviour > behaviourMax { //limit behaviour to max swing
+		a.currBehaviour = behaviourMax
+	} else if newBehaviour < behaviourMin {
+		a.currBehaviour = behaviourMin
+	} else {
+		a.currBehaviour = newBehaviour
+	}
+
```

Comment:
```suggestion
       	a.currBehaviour = behaviour(math.Max(float64(behaviourMin), math.Min(float64(newBehaviour), float64(behaviourMax))))
```

Reply:
Not actually sure if this suggestion is better... I'll think about it some more tomorrow

Reply:
Okay I think the original is more readable, ignore the suggestion here.